File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-204934-85557/analysis/own-tests-project/src/main/java/surveycrawler/FirmSand.java
package surveycrawler;

public class FirmSand extends TraversableGround {

    // moving

    @Override
    public Point positionAfterMovingFrom(Point aPosition, Point aDirection) {
        return aPosition.plus(aDirection);
    }

    @Override
    public Point predictablePositionAfterMovingFrom(Point aPosition, Point aDirection) {
        return positionAfterMovingFrom(aPosition, aDirection);
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/analysis/acceptance-project/src/surveycrawler/firm_sand.py
from surveycrawler.traversable_ground import TraversableGround


class FirmSand(TraversableGround):
    """The crawler moves over firm sand exactly one position per command."""

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.displace(a_direction)
